gremlin-dotnet/src/Gremlin.Net/Driver/Connection.cs

```diff
@@ -68,18 +71,54 @@ public async Task ConnectAsync()
 
         public async Task CloseAsync()
         {
+            if (!String.IsNullOrEmpty(_sessionId))
+            {
+                await closeSession().ConfigureAwait(false);
+            }
             await _webSocketConnection.CloseAsync().ConfigureAwait(false);
         }
 
+        private async Task closeSession()
+        {
+            var msgBuilder = RequestMessage.Build(Tokens.OpsClose).Processor(Tokens.ProcessorSession);
+            msgBuilder.AddArgument(Tokens.ArgsGremlin, "session.close()").AddArgument("session", _sessionId);
+
+            var graphsonMsg = _graphSONWriter.WriteObject(msgBuilder.Create());
```

Comment:
Any reason why you didn't simply call `SendAsync` here? I think it would be good if we wouldn't have to duplicate the code to serialize and send a message.

Reply:
ok, thanks